pkg/tsdb/influxdb/response_parser.go

```diff
@@ -62,26 +62,45 @@ func transformRows(rows []Row, query Query) data.Frames {
 			}
 
 			var timeArray []time.Time
-			var valueArray []*float64
+			var valType = typeof(row.Values[0][columnIndex])
+			var floatArray []*float64
```

Comment:
one suggestion might be to add separate functions for each type:

```golang
func parseStringSeries(row) []*data.Field {}
func parseFloatSeries(row) []*data.Field {}
func parseBoolSeries(row) []*data.Field {}
```

then you can use `make` in each function to set the length of the slices without allocating all three types. something like this:

```golang
func parseStringSeries(row) []*data.Field {
  timeArray := make([]time.Time, len(row.Values))
  floatArray := make([]*float64, len(row.Values))
  for i, valuePair := range row.Values {
    if timestamp, err := parseTimestamp(valuePair[0]); err != nil {
      timeArray[i] = timestamp
      floatArray[i] = parseNumber(valuePair[columnIndex])
    }
  }
  timeField := data.NewField("time", nil, timeArray)
  valueField := data.NewField("value", row.Tags, floatArray)
  return []*data.Field { timeField, valueField }
}

Reply:
i should have read more of the code before writing this suggestion, but i just noticed that it looks like this data seems to be sent as rows. using a wide format time series instead of the multi-frame series might be a better fit for this data since the time column would only need to be parsed once